docs/source/distributed.pipelining.rst

```diff
@@ -4,139 +4,216 @@
 Pipeline Parallelism
 ####################
 
-.. note:: ``torch.distributed.pipelining`` is a package migrated from the `PiPPy <https://github.com/pytorch/PiPPy>`_ project. It is currently in alpha state and under extensive development. For examples that work with our APIs, please refer to PiPPy's `examples <https://github.com/pytorch/PiPPy/tree/main/examples>`_ directory.
+.. note:: ``torch.distributed.pipelining`` is currently in alpha state and under
+development. It is a package migrated from the `PiPPy <https://github.com/pytorch/PiPPy>`_ project.
+
 
 Why Pipeline Parallel?
 **********************
 
-One of the most important techniques for advancing the state of the art in deep learning is scaling. Common techniques for scaling neural networks include *data parallelism*, *tensor/operation parallelism*, and *pipeline parallelism* (or *pipelining*). Pipelining is a technique in which the *code* of the model is partitioned and multiple *micro-batches* execute different parts of the model code concurrently. In many cases, pipeline parallelism can be an effective technique for scaling, in particular for large-scale jobs or bandwidth-limited interconnects. To learn more about pipeline parallelism in deep learning, see `this article <https://www.deepspeed.ai/tutorials/pipeline/>`_.
+One of the most important techniques for advancing the state of the art in deep
+learning is scaling.
+Common techniques for scaling neural networks include *data parallelism*,
+*tensor/operation parallelism*, and *pipeline parallelism*.
+Pipelining is a technique in which the *code* of the model is partitioned and
+multiple *micro-batches* execute different parts of the model code concurrently.
+In many cases, pipeline parallelism can be an effective technique for scaling,
+in particular for large-scale jobs or bandwidth-limited interconnects.
+To learn more about pipeline parallelism in deep learning, see
+`this article <https://www.deepspeed.ai/tutorials/pipeline/>`_.
+
 
 What is ``torch.distributed.pipelining``?
 *****************************************
 
-.. automodule:: torch.distributed.pipelining
+While promising for scaling, pipelining is often difficult to implement,
+requiring intrusive code changes to model code and difficult-to-implement
+runtime orchestration code.
+The ``pipelining`` package aims to provide **a
+toolkit that does said things automatically to allow high-productivity
+implementation of pipeline parallelism on general models.**
 
-.. currentmodule:: torch.distributed.pipelining
+The ``pipelining`` package consists of two parts: a
+*transformation frontend* and a *runtime*.
+The transformation frontend takes your model code, splits it up, and transforms
+it into a ``Pipe``, which describes the "model partition" at each pipeline stage
```

Comment:
the Pipe is only part of the tracer frontend right? so we should only mention it in the tracer section.  Maybe describe that the PipelineStage object is the output, representing and wrapping the portion of the model that will run in this logical stage